### PR TITLE
NANO configuration Cleanup

### PR DESCRIPTION
**Pull request by vlimant**
#### PR description:

what we intend to put in this PR is a cleanup of the nano configuration:

- [x] - drop support of old mini input (drop<=10.2.X)
- [x] - include module validation through fillDescription
- [x] - normalizing of modifiers syntax ( #39342 )

#### PR validation:

no change of output is expected

`runTheMatrix.py --what nano --site "" --command "--number 10"` is the standard candle.

cms-bot nano validation using https://github.com/cms-sw/cms-bot/pull/1867 would be great.

### CODE REVIEW

**Comment by vlimant** [comment]
@perrotta ; is there a place I can read more about how to "safely clone" modules ? it's sort of easy for simple types, but `PSet` and `VPSet` are less trivial

**Comment by perrotta** [comment]
> @perrotta ; is there a place I can read more about how to "safely clone" modules ? it's sort of easy for simple types, but `PSet` and `VPSet` are less trivial

@vlimant you can use python dictionaries [1] to replace existing parameters in PSets, or VPSets.
As of myself, if the "safe clone" gets too much complicate, as it could be the case for deeply nested PSets, I get rid of it and I stick on the "unsafe" mode: it is clear that some extra care and checks, is needed: in particular when the "mother" config has to be modified, one has to carefully check all cloned ones that the same parameter names are correctly propagated.

[1] e.g. https://docs.python.org/3/tutorial/datastructures.html#dictionaries

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
@cms-sw/trk-dpg-l2 @cms-sw/alca-l2  should we bring SimpleTrackFlatTableProducer under PhysicsTools/NanoAOD ? I'll fix the linking error otherwise

File: PhysicsTools/NanoAOD/python/btagWeightTable_cff.py [hunk 1]
```diff
@@ -1,7 +1,5 @@
 import FWCore.ParameterSet.Config as cms
 from PhysicsTools.NanoAOD.common_cff import *
-from PhysicsTools.NanoAOD.nano_eras_cff import *
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/btagWeightTable_cff.py:3]: We discussed removing these branches altogether, perhaps it could be done in this PR? (though unfortunately we have to keep the producer code for V10 compat)

File: PhysicsTools/NanoAOD/plugins/NanoAODBaseCrossCleaner.cc [hunk 2]
```diff
@@ -70,30 +70,33 @@ void NanoAODBaseCrossCleaner::produce(edm::Event& iEvent, const edm::EventSetup&
...
     eles.push_back(electronSel_(e));
   }
   auto electronsTable = std::make_unique<nanoaod::FlatTable>(electronsProd.size(), electronName_, false, true);
 
-  edm::Handle<edm::View<pat::Electron>> lowPtElectronsIn;
   std::vector<uint8_t> lowPtEles;
   if (!lowPtElectronsTag_.label().empty()) {
-    iEvent.getByToken(lowPtElectrons_, lowPtElectronsIn);
-    for (const auto& e : *lowPtElectronsIn) {
+    const auto& lowPtelectronsProd = iEvent.get(lowPtElectrons_);
```
Comment (vlimant) [inline PhysicsTools/NanoAOD/plugins/NanoAODBaseCrossCleaner.cc:97]: to go along #39689

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
@perrotta : I want to pass on the "Safe cloning" for now, as this PR is already quite large and the gain of that change, given the full parameter validation included is quite small. 
I edited out "- safer cloning mechanism" from the PR description

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
please test

**Comment by perrotta** [comment]
> @perrotta : I want to pass on the "Safe cloning" for now, as this PR is already quite large and the gain of that change, given the full parameter validation included is quite small. I edited out "- safer cloning mechanism" from the PR description

@vlimant it's funny that you get scared by something which is far simpler than the other updates and cleanups that you have applied here and in other PRs of yours,
In any case, that's fine with me, of course: please go ahead as you have planned.

**Comment by vlimant** [comment]
how does one test `136.72412` locally since it seems that the first input file is not on disk 
https://cmsweb.cern.ch/das/request?view=list&limit=50&instance=prod%2Fglobal&input=site+file%3D%2Fstore%2Fdata%2FRun2016B%2FJetHT%2FAOD%2F21Feb2020_ver2_UL2016_HIPM-v1%2F120001%2F009C68F1-0F74-DA4E-836B-737D1298A01A.root
maybe we need a matrix option to read IB test files ? are they on EOS somewhere @smuzaffar ?

**Comment by vlimant** [comment]
@cms-sw/reconstruction-l2 : you might be interested in changes in [PhysicsTools/SelectorUtils](https://github.com/cms-sw/cmssw/pull/39796#diff-23e00b17339f24d02f616e9a81ff6a0d69155d1fc41c994fb1cff78e5c812a03)
@cms-sw/alca-l2 @cms-sw/operations-l2 @cms-sw/pdmv-l2 @cms-sw/upgrade-l2 : please let me know your feedback

**Comment by vlimant** [comment]
please test

**Comment by smuzaffar** [comment]
> how does one test `136.72412` locally since it seems that the first input file is not on disk https://cmsweb.cern.ch/das/request?view=list&limit=50&instance=prod%2Fglobal&input=site+file%3D%2Fstore%2Fdata%2FRun2016B%2FJetHT%2FAOD%2F21Feb2020_ver2_UL2016_HIPM-v1%2F120001%2F009C68F1-0F74-DA4E-836B-737D1298A01A.root maybe we need a matrix option to read IB test files ? are they on EOS somewhere @smuzaffar ?

@vlimant , for local tests, please use `--ibeos` [option of runTheMatrix.py](https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/scripts/runTheMatrix.py#L305) . This uses a special site configuration to first search `/eos/cms/store/user/cmsbuild` for root files

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
enable nano

File: PhysicsTools/NanoAOD/plugins/GlobalVariablesTableProducer.cc [hunk 3]
```diff
@@ -40,6 +43,26 @@ class GlobalVariablesTableProducer : public edm::stream::EDProducer<> {
...
+    desc.addOptional<std::string>("name")->setComment("name of the branch in the flat table output");
+    desc.addOptional<bool>("extension", false)->setComment("whether or not to extend an existing same table");
+    edm::ParameterSetDescription variable;
+    variable.ifValue(edm::ParameterDescription<std::string>(
+                         "type", "int", true, edm::Comment("the c++ type of the branch in the flat table")),
+                     edm::allowedValues<std::string>(
+                         "int", "float", "double", "bool", "candidatescalarsum", "candidatesize", "candidatesummass"));
+    variable.add<edm::InputTag>("src")->setComment("input collection for the branch");
+    variable.add<std::string>("doc")->setComment("few words description of the branch content");
+    variable.addOptional<int>("precision")->setComment("precision to store the information [NOT USED IN THE CODE]");
```
Comment (vlimant) [inline PhysicsTools/NanoAOD/plugins/GlobalVariablesTableProducer.cc:57]: N.B. this is added to avoid breaking existing configurations specifying `precision` even though it is not used. a subsequent PR will add the usage of precision for `GlobalVariablesTableProducer`

File: PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h [hunk 4]
```diff
@@ -266,6 +113,82 @@ class PFJetIDSelectionFunctor : public Selector<pat::Jet> {
...
+                                                 "SUMMER18PUPPI",
+                                                 "RUN2UL16CHS",
+                                                 "RUN2UL16PUPPI",
+                                                 "RUN2ULCHS",
+                                                 "RUN2ULPUPPI"));
+    desc.ifValue(edm::ParameterDescription<std::string>("quality", "TIGHT", true, edm::Comment("")),
+                 edm::allowedValues<std::string>("LOOSE", "TIGHT", "TIGHTLEPVETO"));
+    desc.addOptional<std::vector<std::string>>("cutsToIgnore")->setComment("");
+
+    edm::ParameterDescription<double> CHF("CHF", true, edm::Comment(""));
```
Comment (vlimant) [inline PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h:138]: N.B. we should have all these with proper comments in a subsequent PR

**Comment by vlimant** [comment]
please test

File: PhysicsTools/NanoAOD/python/V10/nano_cff.py [hunk 5]
```diff
@@ -179,16 +175,16 @@ def nanoAOD_recalibrateMETs(process,isData):
...
+    #run2_nanoAOD_106Xv1.toModify(process.puppi, useExistingWeights = False)
     print("will make Puppies on top of MINIAOD")
 
 #    makePuppiesFromMiniAOD(process,True) # call this before in the global customizer otherwise it would reset photon IDs in VID
     nanoAOD_PuppiV15_switch = cms.PSet(
             recoMetFromPFCs = cms.untracked.bool(False),
             reclusterJets = cms.untracked.bool(False),
             )
-    run2_nanoAOD_106Xv1.toModify(nanoAOD_PuppiV15_switch,recoMetFromPFCs=True,reclusterJets=True)
+    #run2_nanoAOD_106Xv1.toModify(nanoAOD_PuppiV15_switch,recoMetFromPFCs=True,reclusterJets=True)
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/V10/nano_cff.py:187]: All this below could in principle also be removed

File: PhysicsTools/NanoAOD/python/btagWeightTable_cff.py (hunk 1 above)
Comment (swertz) [reply PhysicsTools/NanoAOD/python/btagWeightTable_cff.py:3]: Reminder

File: PhysicsTools/NanoAOD/python/common_cff.py [hunk 6]
```diff
@@ -12,9 +12,9 @@ def OVar(valtype, compression=None, doc=None, mcOnly=False,precision=-1):
     elif valtype == bool:  valtype = "bool"
     return cms.PSet( 
                 type = cms.string(valtype),
-                compression = cms.string(compression if compression else "none"),
+                #compression = cms.string(compression if compression else "none"),
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/common_cff.py:15]: If these two are not used in C++ don't we want to remove them entirely from there?

File: PhysicsTools/NanoAOD/python/electrons_cff.py [hunk 7]
```diff
@@ -428,70 +284,8 @@ def _get_bitmapVIDForEle_docstring(modules,WorkingPoints):
         dEsigmaDown=Var("userFloat('ecalTrkEnergyPostCorrNew')-userFloat('energySigmaDownNew')", float,  doc="ecal energy smearing value shifted 1 sigma up", precision=8),
 
 )
-##Keeping the possibilty of using V1 working points in older eras
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/electrons_cff.py:431]: All the v1 cutbased and MVA IDs userFloats and userInts can now be removed as well from the `slimmedElectronsWithUserData` above

File: PhysicsTools/NanoAOD/python/nano_cff.py [hunk 8]
```diff
@@ -224,33 +156,15 @@ def nanoAOD_activateVID(process):
     for modname in electron_id_modules_WorkingPoints_nanoAOD.modules:
         setupAllVIDIdsInModule(process,modname,setupVIDElectronSelection)
 
-    electronTask_ = process.egmGsfElectronIDTask.copy()
-    electronTask_.add(electronTask.copy())
-    process.electronTask = electronTask_.copy()
-    for modifier in run2_miniAOD_80XLegacy,run2_nanoAOD_94XMiniAODv1,run2_nanoAOD_94XMiniAODv2,run2_nanoAOD_94X2016,run2_nanoAOD_102Xv1,run2_nanoAOD_106Xv1:
-        modifier.toModify(process.electronMVAValueMapProducer, src = "slimmedElectronsUpdated")
-        modifier.toModify(process.egmGsfElectronIDs, physicsObjectSrc = "slimmedElectronsUpdated")
+    process.electronTask.add( process.egmGsfElectronIDTask )
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/nano_cff.py:159]: For my education, when is a copy of the task needed and when not?

File: PhysicsTools/NanoAOD/python/nano_cff.py [hunk 9]
```diff
@@ -276,32 +190,33 @@ def nanoAOD_customizeCommon(process):
     nanoAOD_tau_switch = cms.PSet(
         idsToAdd = cms.vstring()
     )
-    (run2_nanoAOD_94XMiniAODv1 | run2_nanoAOD_94X2016 | run2_nanoAOD_94XMiniAODv2 | run2_nanoAOD_102Xv1 | run2_nanoAOD_106Xv1).toModify(nanoAOD_tau_switch, idsToAdd = ["deepTau2017v2p1"])
-    (run2_nanoAOD_94XMiniAODv1 | run2_nanoAOD_94X2016 | run2_nanoAOD_94XMiniAODv2 | run2_nanoAOD_102Xv1 | run2_nanoAOD_106Xv1).toModify(process, lambda p : nanoAOD_addTauIds(p, nanoAOD_tau_switch.idsToAdd.value()))
-    (run2_nanoAOD_106Xv2 | run3_nanoAOD_122).toModify(nanoAOD_tau_switch, idsToAdd = ["deepTau2018v2p5"])
-    (run2_nanoAOD_106Xv2 | run3_nanoAOD_122).toModify(process, lambda p : nanoAOD_addTauIds(p, nanoAOD_tau_switch.idsToAdd.value()))
+    #(run2_nanoAOD_106Xv1).toModify(
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/nano_cff.py:193]: some more comments to be removed

File: PhysicsTools/NanoAOD/python/nano_cff.py [hunk 10]
```diff
@@ -276,32 +190,33 @@ def nanoAOD_customizeCommon(process):
...
+    ).toModify(
+        process, lambda p : nanoAOD_addBoostedTauIds(p, nanoAOD_boostedTau_switch.idsToAdd.value())
+    )
 
     return process
 
 def nanoAOD_customizeData(process):
     process = nanoAOD_customizeCommon(process)
-    for modifier in run2_miniAOD_80XLegacy,run2_nanoAOD_94X2016,run2_nanoAOD_94XMiniAODv1,run2_nanoAOD_94XMiniAODv2,run2_nanoAOD_102Xv1,run2_nanoAOD_106Xv1:
-        modifier.toModify(process, lambda p: nanoAOD_recalibrateMETs(p,isData=True))
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/nano_cff.py:294]: I think this solves the item https://github.com/cms-sw/cmssw/issues/34443 we had in our todo list?

File: PhysicsTools/NanoAOD/python/nano_cff.py [hunk 11]
```diff
@@ -68,9 +70,7 @@
 )
 
 
-lhcInfoTable = cms.EDProducer("LHCInfoProducer",
-    precision = cms.int32(10),
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/nano_cff.py:72]: this can be also removed from nanogen_cff.py

File: PhysicsTools/NanoAOD/python/photons_cff.py [hunk 12]
```diff
@@ -81,40 +75,26 @@ def make_bitmapVID_docstring(id_modules_working_points_pset):
...
-    correctionFile = cms.string("EgammaAnalysis/ElectronTools/data/ScalesSmearings/Run2017_17Nov2017_v1_ele_unc")
-)
-
-
-run2_miniAOD_80XLegacy.toModify(calibratedPatPhotonsNano,
-                                correctionFile = cms.string("EgammaAnalysis/ElectronTools/data/ScalesSmearings/Legacy2016_07Aug2017_FineEtaR9_v3_ele_unc")
-)
-
-
 slimmedPhotonsWithUserData = cms.EDProducer("PATPhotonUserDataEmbedder",
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/photons_cff.py:98]: Contrary to the electrons, it seems that the V1 cutBased ID is still used in the output table, but we might want to remove it... perhaps @cms-sw/egamma-pog-l2 can comment?

File: PhysicsTools/NanoAOD/python/simpleSingletonCandidateFlatTableProducer_cfi.py [hunk 13]
```diff
@@ -0,0 +1,8 @@
+import FWCore.ParameterSet.Config as cms
+
+from PhysicsTools.NanoAOD.simpleCandidateFlatTableProducer_cfi import simpleCandidateFlatTableProducer as default
+
+simpleSingletonCandidateFlatTableProducer = default.clone(
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/simpleSingletonCandidateFlatTableProducer_cfi.py:5]: why is this needed?

File: PhysicsTools/NanoAOD/python/taus_updatedMVAIds_cff.py [hunk 14]
```diff
@@ -6,7 +6,7 @@
 from RecoTauTag.RecoTau.PATTauDiscriminationByMVAIsolationRun2_cff import *
 from RecoTauTag.RecoTau.TauDiscriminatorTools import noPrediscriminants
 
-from PhysicsTools.NanoAOD.nano_eras_cff import *
+#from PhysicsTools.NanoAOD.nano_eras_cff import *
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/taus_updatedMVAIds_cff.py:9]: can be removed

File: PhysicsTools/NanoAOD/python/taus_updatedMVAIds_cff.py [hunk 15]
```diff
@@ -281,162 +281,25 @@
...
 for m in patTauDiscriminationByElectronRejectionMVA62015.mapping:
     m.cut = m.cut.value().replace(antiElectronDiscrMVA6_version, antiElectronDiscrMVA6v1_version + "_gbr")
-### Put all anti-e tau-IDs into a sequence
-_patTauDiscriminationByElectronRejection2018Task = cms.Task(
-    patTauDiscriminationByElectronRejectionMVA62018Raw,
-    patTauDiscriminationByElectronRejectionMVA62018
-)
-_patTauDiscriminationByElectronRejection2015Task = cms.Task(
+
+patTauDiscriminationByElectronRejectionTask = cms.Task(
```
Comment (swertz) [inline PhysicsTools/NanoAOD/python/taus_updatedMVAIds_cff.py:285]: Are the "2015" IDs still used anywhere?

**Review by swertz — CHANGES_REQUESTED** [review]

File: PhysicsTools/NanoAOD/python/simpleSingletonCandidateFlatTableProducer_cfi.py (hunk 13 above)
Comment (vlimant) [reply PhysicsTools/NanoAOD/python/simpleSingletonCandidateFlatTableProducer_cfi.py:5]: the default cgi is for `singleton=false `that requires `cut` to be specified. cloning with `singleton=True ` would pul in `cut=""` and a parameter validation error. `simpleCandidateFlatTableProducer_cfi` makes it proper default parameter (i.e. without `cut` defined at all)

File: PhysicsTools/NanoAOD/plugins/NanoAODBaseCrossCleaner.cc [hunk 16]
```diff
@@ -70,30 +70,33 @@ void NanoAODBaseCrossCleaner::produce(edm::Event& iEvent, const edm::EventSetup&
...
   std::vector<uint8_t> eles;
+  eles.reserve(electronsProd.size());
   for (const auto& e : electronsProd) {
     eles.push_back(electronSel_(e));
   }
   auto electronsTable = std::make_unique<nanoaod::FlatTable>(electronsProd.size(), electronName_, false, true);
 
-  edm::Handle<edm::View<pat::Electron>> lowPtElectronsIn;
   std::vector<uint8_t> lowPtEles;
   if (!lowPtElectronsTag_.label().empty()) {
```
[the remark this answers is not in the document]
Comment (vlimant) [reply PhysicsTools/NanoAOD/plugins/NanoAODBaseCrossCleaner.cc:96]: indeed.

File: PhysicsTools/NanoAOD/plugins/NanoAODDQM.cc [hunk 17]
```diff
@@ -206,6 +212,52 @@ NanoAODDQM::NanoAODDQM(const edm::ParameterSet &iConfig) {
...
+  edm::ParameterSetDescription plot;
+  plot.setComment("a parameter set that defines a DQM histogram");
+  plot.ifValue(
+      edm::ParameterDescription<std::string>("kind", "none", true, edm::Comment("the type of histogram")),
+      "none" >> (name) or  //it should really be edm::EmptyGroupDescription(), but name is used in python by modifiers
+          "count1d" >> (name and title and nbins and min and max) or
+          "hist1d" >> (name and title and nbins and min and max and column) or
+          "prof1d" >> (name and title and nbins and min and max and xcolumn and ycolumn));
+
+  //edm::ParameterSetDescription plots;
```
[the remark this answers is not in the document]
Comment (vlimant) [reply PhysicsTools/NanoAOD/plugins/NanoAODDQM.cc:243]: removed

File: PhysicsTools/NanoAOD/python/nano_cff.py (hunk 9 above)
Comment (vlimant) [reply PhysicsTools/NanoAOD/python/nano_cff.py:193]: yup

File: PhysicsTools/NanoAOD/python/nano_cff.py (hunk 10 above)
Comment (vlimant) [reply PhysicsTools/NanoAOD/python/nano_cff.py:294]: maybe, I believe there is another place where this gets in

File: PhysicsTools/NanoAOD/python/nano_cff.py (hunk 11 above)
Comment (vlimant) [reply PhysicsTools/NanoAOD/python/nano_cff.py:72]: I don't see `LHCInfoProducer ` in nanogen

File: PhysicsTools/NanoAOD/python/taus_updatedMVAIds_cff.py (hunk 15 above)
Comment (vlimant) [reply PhysicsTools/NanoAOD/python/taus_updatedMVAIds_cff.py:285]: maybe not. we should come back to this later on though

File: PhysicsTools/NanoAOD/python/taus_updatedMVAIds_cff.py (hunk 14 above)
Comment (vlimant) [reply PhysicsTools/NanoAOD/python/taus_updatedMVAIds_cff.py:9]: right

File: PhysicsTools/NanoAOD/python/nano_cff.py (hunk 8 above)
Comment (vlimant) [reply PhysicsTools/NanoAOD/python/nano_cff.py:159]: beats me :-)

File: PhysicsTools/NanoAOD/python/common_cff.py (hunk 6 above)
Comment (vlimant) [reply PhysicsTools/NanoAOD/python/common_cff.py:15]: was considering indeed ; done

File: PhysicsTools/NanoAOD/python/muons_cff.py [hunk 18]
```diff
@@ -14,16 +16,13 @@
     miniIsoParams = PhysicsTools.PatAlgos.producersLayer1.muonProducer_cfi.patMuons.miniIsoParams, # so they're in sync
     recomputeMuonBasicSelectors = cms.bool(False),
 )
-run2_miniAOD_80XLegacy.toModify( slimmedMuonsUpdated, computeMiniIso = True, recomputeMuonBasicSelectors = True )
```
[the remark this answers is not in the document]
Comment (vlimant) [reply PhysicsTools/NanoAOD/python/muons_cff.py:17]: note taken

File: PhysicsTools/NanoAOD/plugins/VertexTableProducer.cc [hunk 19]
```diff
@@ -202,11 +202,23 @@ void VertexTableProducer::endStream() {}
...
+  desc.add<edm::InputTag>("svSrc")->setComment(
+      "reco::VertexCompositePtrCandidate compatible secondary vertex input collection");
+  desc.add<std::string>("svCut")->setComment("selection on the secondary vertex");
+
+  desc.add<double>("dlenMin")->setComment("minimum value of dl to select secondary vertex");
+  desc.add<double>("dlenSigMin")->setComment("minimum value of dl significance to select secondary vertex");
+
+  desc.add<std::string>("pvName")->setComment("name of the flat table ouput");
+  desc.add<std::string>("svName")->setComment("name of the flat table ouput");
+  desc.add<std::string>("svDoc")->setComment("a few words of documentation [NOT USED IN THE CODE]");
```
[the remark this answers is not in the document]
Comment (vlimant) [reply PhysicsTools/NanoAOD/plugins/VertexTableProducer.cc:219]: added as `setDoc`

File: Configuration/PyReleaseValidation/python/relval_nano.py [hunk 20]
```diff
@@ -244,74 +226,64 @@ def subnext(self):
                             _NANO_data])
 
 
-_wfn=WFN()
 ################
+"""
 #8.X input
-workflows[_wfn()] = ['mc80X', ['TTbarMINIAOD8.0','NANO_mc8.0', 'HRV_NANO_mc']]
-workflows[_wfn()] = ['data80X', ['MuonEG2016HMINIAOD8.0','NANO_data8.0', 'HRV_NANO_data']]
+workflows[2500.0] = ['mc80X', ['TTbarMINIAOD8.0','NANO_mc8.0', 'HRV_NANO_mc']]
```
[the remark this answers is not in the document]
Comment (vlimant) [reply Configuration/PyReleaseValidation/python/relval_nano.py:232]: gone to git history

File: PhysicsTools/NanoAOD/python/electrons_cff.py (hunk 7 above)
Comment (vlimant) [reply PhysicsTools/NanoAOD/python/electrons_cff.py:431]: noted

File: PhysicsTools/NanoAOD/python/nano_cff.py (hunk 10 above)
Comment (swertz) [reply PhysicsTools/NanoAOD/python/nano_cff.py:294]: Do you know where?

File: PhysicsTools/NanoAOD/python/nano_cff.py (hunk 11 above)
Comment (swertz) [reply PhysicsTools/NanoAOD/python/nano_cff.py:72]: My bad, I misread `lhcInfoTable` and `lheIntoTable`

**Comment by vlimant** [comment]
is there any feedback from @cms-sw/alca-l2 @cms-sw/reconstruction-l2 @cms-sw/pdmv-l2 @cms-sw/operations-l2 @cms-sw/upgrade-l2 on this ? or please sign at earliest convenience.

**Comment by vlimant** [comment]
please test

**Comment by tvami** [comment]
+alca
 * alca change is in BuildFiles only
 * tests pass

**Comment by perrotta** [comment]
ping @cms-sw/reconstruction-l2 @cms-sw/pdmv-l2 @cms-sw/upgrade-l2

**Comment by perrotta** [comment]
@vlimant I think that if this gets signed by @cms-sw/xpog-l2 other people can get convinced that the PR is considered final by the author and it is ready for review and/or signature

**Comment by tvami** [comment]
hi @vlimant while we wait for other groups to review the code changes, I'm wondering if it would be beneficial to squash the commits that are like "fixing all typos", "fix further typos", "code cleaning", several "code formatting" commits, "removing commented lines", etc, you know what I mean. Thanks

**Comment by vlimant** [comment]
will this change anything @tvami ?

**Comment by srimanob** [comment]
Hi @vlimant 

Would you mind to provide a bit more information on the strategy of Nano dev? For example, I see that you drop 
`'run3_nanoAOD_devel'` and keep only `'run3_nanoAOD_122', 'run3_nanoAOD_124'`. What will happen when developers would like to merge new features to `NANO`? Which era that new features will stay with? Thanks.

**Comment by vlimant** [comment]
Hi @srimanob ,
new feature will go in master and developing release. modifier based integration is not a good way to do things IMO, and turned out to make maintenance unbearable. New nano version for wider usage (production or else) will be managed like the V10.

**Comment by srimanob** [comment]
+Upgrade

From upgrade related config, this PR clean up old config and change the strategy how to deal with dev in Nano.

**Comment by vlimant** [comment]
@cms-sw/reconstruction-l2 @cms-sw/pdmv-l2 I hope we can get a signature this week

**Comment by mandrenguyen** [comment]
+1

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
@vlimant since the merge of this PR there are several crashes in the nanoAOD workflows in the IB wit the message

```
Traceback (most recent call last):
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02758/el8_amd64_gcc10/cms/cmssw/CMSSW_12_6_X_2022-11-08-1100/bin/el8_amd64_gcc10/cmsDriver.py", line 51, in <module>
    run()
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02758/el8_amd64_gcc10/cms/cmssw/CMSSW_12_6_X_2022-11-08-1100/bin/el8_amd64_gcc10/cmsDriver.py", line 28, in run
    configBuilder.prepare()
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02758/el8_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-11-08-2000/src/Configuration/Applications/python/ConfigBuilder.py", line 2325, in prepare
    self.pythonCfgCode += self.addCustomise()
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02758/el8_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-11-08-2000/src/Configuration/Applications/python/ConfigBuilder.py", line 896, in addCustomise
    __import__(packageName)
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02758/el8_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-11-08-2000/src/PhysicsTools/NanoAOD/python/custom_jme_cff.py", line 12, in <module>
    from PhysicsTools.NanoAOD.jetsAK4_CHS_cff import jetTable, jetCorrFactorsNano, updatedJets, finalJets, qgtagger, hfJetShowerShapeforNanoAOD
ImportError: cannot import name 'hfJetShowerShapeforNanoAOD' from 'PhysicsTools.NanoAOD.jetsAK4_CHS_cff' (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02758/el8_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-11-08-2000/src/PhysicsTools/NanoAOD/python/jetsAK4_CHS_cff.py)
```

Can you please have a look?

**Comment by vlimant** [comment]
sure, how can I reproduce this ?

**Comment by perrotta** [comment]
> sure, how can I reproduce this ?

By running the failing workflows: 
10224.15
11024.15
11634.15
25202.15

**Comment by vlimant** [comment]
we'll do that as soon as there's an IB available ; we do not find `CMSSW_12_6_X_2022-11-09*` for now
